lib/galaxy/datatypes/converters/archive_to_directory.xml

```diff
@@ -0,0 +1,33 @@
+<tool id="CONVERTER_archive_to_directory" name="Unpack archive to directory" version="1.0.0" profile="21.09">
+    <!-- Use compression_utils instead of shell commands (tar/unzip) so we can verify safety of results -->
+    <requirements>
+        <requirement type="package" version="23.2.1">galaxy-util</requirement>
+    </requirements>
+    <command><![CDATA[
+        mkdir '$output1.files_path' &&
+        cd '$output1.files_path' &&
+        python -c "from galaxy.util.compression_utils import CompressedFile; CompressedFile('$input1').extract('.')"
+    ]]></command>
+    <configfiles>
+        <configfile filename="metadata_json"><![CDATA[{
+    "output1": {
+        "name": "$input1.name unpacked to $__target_datatype__",
+        "ext": "$__target_datatype__"
+    }
+}]]></configfile>
+    </configfiles>
+    <inputs>
+        <param format="tar,zip" name="input1" type="data"/>
```

Comment:
Should we add `tar.gz` or `tar.bz2`?

Edit: seems that we do not have these datatypes yet, or?